spmd/tensor/parallel/api.py

```diff
@@ -76,12 +76,75 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 module.register_module(n, tp_multi_head_attention)
 
 
+def _is_mlp(module: nn.Module) -> bool:
+    # We assume that the structure of MLP to shard defined in object way as below:
+    #   * -> (Linear -> * -> Linear)+ -> *
+    # positive even number of Linear layers interleaved by activation/norm/dropout
+    linear_submodules = list(
+        filter(lambda x: isinstance(x, nn.Linear), module.children())
+    )
+    return len(linear_submodules) > 0 and len(linear_submodules) % 2 == 0
+
+
+def _shard_mlp(name: str, module: nn.Module, device_mesh: DeviceMesh) -> None:
+    if _is_mlp(module):
+        col_wise_sharding: Sequence[Placement] = [Shard(0)]
+        row_wise_sharding: Sequence[Placement] = [Shard(1)]
+        replicate: Sequence[Placement] = [Replicate()] * device_mesh.ndim
+        linear_submodules = list(
+            filter(lambda x: isinstance(x, nn.Linear), module.children())
+        )
+
+        for i, m in enumerate(linear_submodules):
+            if i % 2 == 0:
+                # shard Linear layer column-wisely
+                sharded_weight = nn.Parameter(
+                    distribute_tensor(
+                        cast(torch.Tensor, m.weight),
+                        device_mesh,
+                        col_wise_sharding,
+                    )
+                )
+                sharded_bias = nn.Parameter(
+                    distribute_tensor(
+                        cast(torch.Tensor, m.bias),
+                        device_mesh,
+                        col_wise_sharding,
+                    )
+                )
```

Comment:
can we make the common logic into a util func? Thanks!

Reply:
or we can just call `_parallelize_linear` instead?

Reply:
That will work.